addons/medical_treatment/ACE_Medical_Treatment_Actions.hpp

```diff
@@ -231,23 +231,33 @@ class GVAR(actions) {
         displayName = CSTRING(Actions_CheckPulse);
         displayNameProgress = CSTRING(Check_Pulse_Content);
         allowedSelections[] = {"All"};
-        condition = QUOTE(GVAR(advancedDiagnose) != 0);
+        treatmentTime = QUOTE([ARR_2(2.5,15)] select GVAR(enableRealisticPulseChecking));
+        condition = QUOTE(GVAR(advancedDiagnose) != 0 && {!GVAR(enableRealisticPulseChecking) || {(_medic getSlotItemName TYPE_WATCH) isKindOf [ARR_2('ItemWatch',configFile >> 'CfgWeapons')]}});
         callbackSuccess = QFUNC(checkPulse);
+        callbackProgress = QFUNC(checkPulseProgress);
+        callbackFailure = QUOTE(QQGVAR(checkPulse) cutText [ARR_2('','PLAIN')]);
         animationMedicProne = "";
         animationMedicSelfProne = "";
     };
     class CheckBloodPressure: CheckPulse {
         displayName = CSTRING(Actions_CheckBloodPressure);
         displayNameProgress = CSTRING(Check_Bloodpressure_Content);
         allowedSelections[] = {"LeftArm", "RightArm", "LeftLeg", "RightLeg"};
+        treatmentTime = 15;
```

Comment:
15 is too long imo. I think 6 would be more reasonable, given the estimates you can do after 6 seconds (I would also change the pulse checking time to 6 too when realistic pulse checking is disabled).